src/app/Shared/Services/Api.service.tsx

```diff
@@ -564,19 +564,33 @@ export class ApiService {
   }
 
   postRecordingMetadataFromPath(jvmId: string, recordingName: string, labels: KeyValue[]): Observable<boolean> {
-    return this.sendRequest(
-      'beta',
-      `fs/recordings/${encodeURIComponent(jvmId)}/${encodeURIComponent(recordingName)}/metadata/labels`,
+    return this.graphql<any>(
+      `
+      query PostRecordingMetadataFromPath($jvmId: String!, $recordingName: String!, $labels: [Entry_String_StringInput]) {
```

Comment:
I think this query (and the function it is within) can be renamed. The `FromPath` part was referring to the old `fs/` part of the old beta HTTP API endpoint, which was always a bit of a hack and now doesn't really refer to anything since the storage is in S3 rather than directly at some path.

Maybe `postRecordingMetadataForJvmId`